ampd/src/monitoring/mod.rs

```diff
@@ -0,0 +1,68 @@
+mod endpoints;
+pub mod server;
+use std::net::{Ipv4Addr, SocketAddrV4};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone)]
+pub enum MetricsMsg {
+    IncBlockReceived,
+}
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, Debug)]
+pub struct Config {
+    pub enabled: bool,
+    pub bind_address: Option<SocketAddrV4>,
```

Comment:
this allows the representation of an invalid state `enabled: true, bind_address: None`. Remove the boolean, it's already covered by the `Option`